routing.md

```diff
@@ -305,12 +321,51 @@ DIDComm DID Document endpoints have the following format:
 
 **type**: MUST be `DIDComm`. 
 
-**serviceEndpoint**: MUST contain a URI for a transport specified in the [transports] section of this spec.
+**serviceEndpoint**: MUST contain a URI for a transport specified in the [transports] section of this spec, or a URI from Alternative Endpoints. It MAY be desirable to constraint endpoints from the [transports] section so that they are used only for the reception of DIDComm messages. This can be particularly helpful in cases where auto-detecting message types is inefficient or undesirable.
 
 **routingKeys**: An optional ordered array of strings referencing keys to be used when preparing the message for transmission as specified in the [Routing] section of this spec. 
 
 #### Multiple Endpoints
 
-A DID Document may contain multiple service entries of type `DIDComm`. Unless otherwise specified, messages should be sent to endpoints in the order they are specified.
+A DID Document may contain multiple service entries of type `DIDComm`. Entries are SHOULD be specified in order of receiver preference, but any endpoint MAY be selected by the sender, typically by protocol availability or preference.
+
+#### Alternative Endpoints
+
+In addition to the URIs for [transports], some alternative forms are available.
+
+##### DID
+
+Using a DID for the `serviceEndpoint` is useful when using a mediator. The DID should be resolved, and services with type of "DIDComm" will contain valid `serviceEndpoints`. The keyAgreement keys of that DID Document should be appended at the end of the routingKeys section from the message recipient's DID Document as per the process in [Sender Forward Process]. The key advantage with this approach is that a mediator can rotate keys and update serviceEndpoints without any updates needed to dependent recipients` DID Documents.
+
+A DID used as a mediator in this way SHOULD NOT use alternative endpoints. Only URIs described in [transports] are should be used to avoid recursive lookups.
```

Comment:
I'm not sure I followed this statement. The way I read and understood it was "A mediator represents their did document in particular ways. One of the ways it SHOULD NOT use it though is to redirect to a different endpoint if it knows it's a mediator service. To avoid this problem mediator services should define their service endpoints in a way that are described in the transports section".

1. Am I understanding this correctly?
2. If so, I'd suggest rewording below:

```suggestion
A DID Document known to represent a mediator service SHOULD NOT use a redirecting endpoint to avoid recursive loops. Rather, it should directly represent it's service endpoints as URIs described in the [transports] section. Additionally, to avoid recursion loops, it's RECOMMENDED that mediator services not configure DID Document recursion layers deeper than 2.
```

Reply:
I see the confusion you point out, but I was also confused a bit by your suggestion. I've reworded this section; please review and see if it satisfies the intent of your comment.